smartlearner/interfaces/optimizer.py

```diff
@@ -79,3 +87,13 @@ def _apply_modifiers(self, list_modifiers, objects_to_modify):
             updates.update(modifier.updates)
 
         return updates
+
+    def save(self, path):
+        # TODO: should we save directions and params modifiers?
```

Comment:
I hate those names so much. I still can't remember what is a param modifiers.
As for your question, I guess we need to save the full internal state. Then again, it depends on when a save is made. If it's inbetween epochs, it might be necessary only for a few optimizer to keep the whole thing.